akka-http-core/src/main/scala/akka/http/scaladsl/Http.scala

```diff
@@ -879,10 +879,32 @@ class HttpExt private[http] (private val config: Config)(implicit val system: Ex
   }
 
   /** Creates real or placebo SslTls stage based on if ConnectionContext is HTTPS or not. */
-  private[http] def sslTlsStage(connectionContext: ConnectionContext, role: TLSRole, hostInfo: Option[(String, Int)] = None) =
+  private[http] def sslTlsClientStage(connectionContext: ConnectionContext, host: String, port: Int) =
```

Comment:
Splitting up made things easier to understand (at least to me) because it was clearer that host/port would be always provided for clients and never for servers. I agree the duplication doesn't make sense, I'll move that back to the original function but make it private.